extensions/amp-ad/0.1/amp-ad.js

```diff
@@ -52,6 +52,9 @@ export class AmpAd extends AMP.BaseElement {
         // Unspecified or empty type.  Nothing to do here except bail out.
         return null;
       }
+      window.ampAdSlotIdCounter = window.ampAdSlotIdCounter || 0;
+      const slotId = window.ampAdSlotIdCounter++;
+      this.element.setAttribute('data-slot-id', slotId);
```

Comment:
Yep, that definitely sounds better.  How about `data-amp-slot-index`? To not be confused with the other slot ID concept.